### PR TITLE
Feature #741: Use opaque cursor on submission list and entity list

### LINKED ISSUE #741: Add infinite scroll to Entities table
For v2023.2, we are just loading the entire block of entities to show in the entity table. For the following release, when we likely add bulk CSV managing of entities, we will want to add the same kind of chunking and infinite scroll that we have for submissions. 

We will likely want to reuse functionality from the submission chunking/scrolling and make a generic infinite scroll odata table. 

Note about infinite scroll vs. paging: one use case we see in practice is loading a bunch of submissions and then searching on the page itself for specific content. This is a pretty good case for sticking with infinite scroll until we have the ability to search submission/entity content.

### PR DESCRIPTION
**Pull request by sadiqkhoja**
Closes #741

#### What has been done to verify that this works as intended?

Integration tests and manual verification

#### Why is this the best possible solution? Were any other approaches considered?

Used the same approach as Submissions List

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

NA

#### Does this change require updates to user documentation? If so, please file an issue [here](https://github.com/getodk/docs/issues/new) and include the link below.

No

#### Before submitting this PR, please make sure you have:

- [x] run `npm run test` and `npm run lint` and confirmed all checks still pass OR confirm CircleCI build passes
- [x] verified that any code or assets from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

**Review by matthew-white — COMMENTED** [review]
This is a great simplification! I love that we no longer need to track instance IDs or have challenges related to scrolling in chunks while there's concurrent creation or deletion. The code here definitely looks like it's on the right track. I've left some suggestions, but I think @ktuite could do the final review without me if that ends up working best. Excited for `$skiptoken` and the changes here!

File: src/components/entity/list.vue
[the remark this answers is not in the document]
Comment (sadiqkhoja) [reply src/components/entity/list.vue:1]: I like the idea but thinking to do this in a separate PR later

File: src/components/entity/list.vue [hunk 1]
```diff
@@ -67,13 +73,25 @@ export default {
     datasetName: {
       type: String,
       required: true
+    },
+    // Returns the value of the $top query parameter.
+    top: {
+      type: Function,
+      default: (loaded) => (loaded < 1000 ? 10 : 1000)
```
Comment (sadiqkhoja) [inline src/components/entity/list.vue:80]: 250

File: src/components/odata-loading-message.vue [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    if (props.filter)
+      return t(`${props.type}.filtered.withoutCount`);
+
+    if (props.totalCount === 0)
+      return t(`${props.type}.withoutCount`);
+
+    if (props.totalCount <= props.top)
+      return tcn(`${props.type}.all`, props.totalCount);
+
+    // console.log( t('submission.first', { count: 'asdf', top: 10 }, 11 ) );
```
Comment (sadiqkhoja) [inline src/components/odata-loading-message.vue:67]: delete this

File: src/util/i18n.js [hunk 3]
```diff
@@ -43,3 +43,8 @@ export const loadLocale = ({ i18n, logger }, locale) => {
 export function $tcn(path, count, values = undefined) {
   return this.$tc(path, count, { count: this.$n(count, 'default'), ...values });
 }
+
+// same as above but for composition API
+export function tcn(path, count, values) {
```
Comment (sadiqkhoja) [inline src/util/i18n.js:48]: double check if I really need this function

**Review by ktuite — APPROVED** [review]
Looked at this after a synchronous review with Sadiq earlier and I think it's good!

**Pull request merged**